include/alpaka/kernel/Traits.hpp

```diff
@@ -83,6 +85,84 @@ namespace alpaka
                 return 0u;
             }
         };
+
+        namespace detail
+        {
+            //#############################################################################
+            //! Functor to get OpenMP schedule defined by kernel class.
+            //! When no schedule is defined, return a default one.
+            template<class TKernel, class = void>
+            struct GetOmpSchedule
+            {
+                ALPAKA_FN_HOST_ACC auto operator()() const
+                {
+                    return alpaka::omp::Schedule{};
+                }
+            };
+
+            //! Functor to get OpenMP schedule for kernel classes with
+            //! ompSchedule static member.
+            //! That member is never odr-used by alpaka.
+            template<class TKernel>
+            struct GetOmpSchedule<TKernel, meta::Void<decltype(TKernel::ompSchedule)>>
+            {
+                ALPAKA_FN_HOST_ACC auto operator()() const
+                {
+                    return TKernel::ompSchedule;
+                }
```

Comment:
Note: here have to go with a functor, as we can't rely that a user-defined member is `constexpr`. And for non-constexpr can't initialize it in place, as it is not a literal type.